yml
- name: Check for gce_pd error with snapshot disks
  hosts: all
  gather_facts: true

  tasks:
  - name: Create a new disk from snapshot
    gce_pd:
      name: "my-snapshot-disk"
      image: "global/snapshots/my-snapshot"
      state: present
    register: disk_creation

  - name: Handle gce_pd error
    fail:
      msg: "Error encountered while creating disk from snapshot: {{ disk_creation.msg }}"
    when: disk_creation.failed
